src/assets/PythonCopyFileWithShutil/python-copy-paste.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Ways to copy file with shutil module in Python
+authorName: Salipa Gurung
+authorAvatar: https://avatars.githubusercontent.com/u/53458341?v=4
+authorLink: https://github.com/Salipa-Gurung
+createdAt: March 11, 2024
+tags: python, copy, shutil
+banner: https://blog.jankaritech.com/src/assets/PythonCopyFileWithShutil/images/python_copy_file.png
+---
+
+Python provides many built-in modules to copy a file. We'll look into ways to copy file using `shutil` module in this blog post.
+
+The standard utility module known as shutil is an abbreviation for shell utilities. Numerous functions in the Python shutil module helps with high-level file and directory operations. Using the shutil module, there are four common methods that provide an easy and efficient way to copy a file:
+
+### 1. shutil.copy() method
+
+```py
+shutil.copy(source, destination, *, follow_symlinks=True)
+```
+
+This method copies the file of source to the directory or file specified in the destination and returns file's destination.
+
+- shutil.copy() method copies file's data and file's permission mode.
+- Metadata(file creation and modification time) of the file copied is not preserved.
+- Source and destination should be path-like objects or strings.
+- Destination can specify both file or directory. If the destination specify path to a directory, the file will be copied to destination directory with the basename of the source path.
+- Copying symbolic link:
+
+  - If `follow_symlinks=True`, destination will be a copy of the file that the symbolic link in source points to. Default value of parameter `follow_symlinks` is `True`.
+
+  - If `follow_symlinks=False`, destination will be created as symbolic link.
+
+  ```py
+  # Import shell utility module
+  import shutil
+
+  #  Copy the content of source_file.txt to destination_file.txt
+  shutil.copy('source_file.txt', 'destination_file.txt')
+  ```
+
+### 2. shutil.copy2() method
+
+```py
+shutil.copy2(source, destination, *, follow_symlinks=True)
+```
+
+The shutil.copy2() method provides additional functionality of preserving all the metadata of the file it can. Other characteristics are same as shutil.copy() method.
```

Comment:
```suggestion
This method provides additional functionality of preserving all the metadata of the file it can. Other characteristics are same as shutil.copy() method.
```